src/main/java/gyro/google/storage/BucketRetentionPolicy.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.storage;
+
+import com.google.api.services.storage.model.Bucket.RetentionPolicy;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.Range;
+import gyro.google.Copyable;
+
+/**
+ * Defines the minimum age an object in the bucket must reach before it can be deleted or overwritten.
+ */
+public class BucketRetentionPolicy extends Diffable implements Copyable<RetentionPolicy> {
+
+    private Long retentionPeriod;
+
+    /**
+     * The duration in seconds that objects need to be retained. Must be greater than ``0`` and less than ``3,155,760,000`` (100 years).
+     */
+    @Updatable
+    @Range(min = 1D, max = 3155759999D)
+    public Long getRetentionPeriod() {
+        return retentionPeriod;
+    }
+
+    public void setRetentionPeriod(Long retentionPeriod) {
+        this.retentionPeriod = retentionPeriod;
+    }
+
+    @Override
+    public String primaryKey() {
+        return "retention-policy-config";
```

Comment:
I'd return `""` for this so the output looks a bit nicer.

Reply:
Done.